src/main/java/com/ait/lienzo/client/core/shape/TextLineBreakWrap.java

```diff
@@ -0,0 +1,106 @@
+/*
+   Copyright (c) 2017 Ahome' Innovation Technologies. All rights reserved.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+ */
+
+package com.ait.lienzo.client.core.shape;
+
+import com.ait.lienzo.client.core.Context2D;
+import com.ait.lienzo.client.core.types.BoundingBox;
+import com.ait.lienzo.shared.core.types.TextAlign;
+import com.ait.lienzo.shared.core.types.TextBaseLine;
+import com.ait.lienzo.shared.core.types.TextUnit;
+
+/**
+ * ITextWrapper implementation that wraps text on line breaks.
+ */
+public class TextLineBreakWrap extends TextNoWrap {
+
+    public TextLineBreakWrap(final Text text) {
+        super(text);
+    }
+
+    public TextLineBreakWrap(final Supplier<String> textSupplier,
+                             final Supplier<Double> fontSizeSupplier,
+                             final Supplier<String> fontStyleSupplier,
+                             final Supplier<String> fontFamilySupplier,
+                             final Supplier<TextUnit> textUnitSupplier,
+                             final Supplier<TextBaseLine> textBaseLineSupplier,
+                             final Supplier<TextAlign> textAlignSupplier) {
+        super(textSupplier,
+              fontSizeSupplier,
+              fontStyleSupplier,
+              fontFamilySupplier,
+              textUnitSupplier,
+              textBaseLineSupplier,
+              textAlignSupplier);
+    }
+
+    @Override
+    public BoundingBox getBoundingBox() {
+        double width = 0;
+        final String text = textSupplier.get();
+        if (text == null || text.isEmpty()) {
+            return new BoundingBox();
+        }
+        final String[] lines = text.split("\\r?\\n");
```

Comment:
Isn't possible to use something like below?
- System.getProperty("line.separator");

Reply:
@jomarko We cannot access System Properties on the client (browser) side.

This Regex is apparently cross-platform friendly. I gleamed it from [here](https://stackoverflow.com/questions/1331815/regular-expression-to-match-cross-platform-newline-characters).